File: src/main/java/model/load/EngineLoad.java
package model.load;

public class EngineLoad {

    // https://mechanics.stackexchange.com/a/17548

    /**
     *
     * @param airflow g/sec
     * @param rpm the engine RPM
     * @param displacement engine displacement in liters
     * @return
     */
    public static double getAbsoluteEngineLoad(double airflow, double rpm, double displacement) {
        double airMass = airflow/((rpm/60d));
        return airMass/(1.184 * displacement)*2;
    }

    /**
     *
     * @param engineLoad %
     * @param rpm the engine RPM
     * @param displacement engine displacement in liters
     * @return
     */
    public static double getAirflow(double engineLoad, double rpm, double displacement) {
        return (engineLoad*(1.184 * displacement)/2)*(rpm/60d);
    }
}

File: src/main/kotlin/domain/model/load/EngineLoad.kt
package domain.model.load

object EngineLoad {
    fun getAbsoluteEngineLoad(airflow: Double, rpm: Double, displacement: Double): Double {
        val airMass = airflow / (rpm / 60.0)
        return airMass / (1.184 * displacement) * 2
    }

    fun getAirflow(engineLoad: Double, rpm: Double, displacement: Double): Double {
        return (engineLoad * (1.184 * displacement) / 2) * (rpm / 60.0)
    }
}
